File: python/KRAKEN/Tests/DW/Munk/MunkKleaky.m
clear all, close all

addpath /home/orodrig/FORdoc/at/Matlab/ReadWrite

disp('KRAKEN - Munk profile, leaky modes') 

fid = fopen('MunkKleaky.sbp','r');
n = fscanf(fid,'%d\n',[1 1]);
data = fscanf(fid,'%f %f\n',[2 n]);
fclose(fid);

angles  = data(1,:); 
powerdB = data(2,:);

figure(1)
polar(angles*pi/180,powerdB)
grid on, box on

system('kraken.exe MunkKleaky');
system('fields.exe MunkKleaky < MunkKleaky.flp');

[ PlotTitle, PlotType, freq, atten, Pos, p ] = read_shd( 'MunkKleaky.shd' );

zs = Pos.s.depth;

rarray = Pos.r.range; rarraykm = rarray/1000;
zarray = Pos.r.depth;

rmax = max( rarray ); rmaxkm = rmax/1000;
zmax = max( zarray );

pressure = squeeze( p );

tl = -20*log10( abs( pressure ) ); 

tej = flipud( jet( 256 ) );

figure(2), hold on
plot(0,zs,'ko',0,zs,'m*','MarkerSize',16) 
pcolor(rarraykm,zarray,tl), shading interp, colormap( tej ), caxis([40 90]), colorbar
axis([0 rmaxkm 0 zmax])
hold off
view(0,-90)
xlabel('Range (km)')
ylabel('Depth (m)')
title('KRAKEN - Munk profile, leaky modes, 50 Hz')

disp('done.')

File: python/KRAKEN/Tests/DW/Munk/MunkKleaky.py
#==================================================================
#  
#  KRAKEN: Munk profile, leaky modes
#  Mexilhoeira Grande, Sab Jul 29 10:36:13 WEST 2017
#  Written by Tordar
#  
#==================================================================

# ipython: run MunkKleaky

from os import system
from numpy import *
from scipy.io import *
from pylab import *
sys.path.append ("/home/orodrig/FORdoc/at/Python/")
from readshd import *

rs = 0.0

print("KRAKEN - Munk profile, leaky modes")

fid = open('MunkKleaky.sbp','r')
theline = fid.readline()
n       = int( theline )
angles  = zeros( n ) 
powerdB = zeros( n )
for i in range(n):
    theline = str( fid.readline() )
    datai = theline.split()
    angles[ i] = float( datai[0] )
    powerdB[i] = float( datai[1] )
fid.close()
anglesRad = angles*pi/180

system("kraken.exe MunkKleaky")
system("fields.exe MunkKleaky < MunkKleaky.flp")

filename = 'MunkKleaky.shd'
xs = nan
ys = nan
pressure,geometry = readshd(filename,xs,ys)

zs     = geometry["zs"]
rarray = geometry["rarray"]; rarraykm = rarray/1000
zarray = geometry["zarray"]

Dmax = zarray[-1]
rmax = rarray[-1]; rmaxkm = rmax/1000

p = squeeze( pressure, axis=(0,1) )
tl = -20*log10( abs( p ) )

figure(1)
polar( anglesRad, abs(powerdB) )

figure(2)
imshow(tl,extent=[0,rmaxkm,0,Dmax],aspect='auto',cmap='jet_r',origin='lower',vmin=40,vmax=90)
cb = colorbar()
cb.ax.invert_yaxis()
plot(rs,-zs,marker="<",markersize=16,color="k")
xlabel('Range (km)')
ylabel('Depth (m)')
title('KRAKEN - Munk profile, leaky modes')
ylim(Dmax,0)

show()

print("done.")
